src/Symfony/Component/Routing/Matcher/UrlMatcher.php

```diff
@@ -259,4 +261,37 @@ protected function createRequest($pathinfo)
             'SCRIPT_NAME' => $this->context->getBaseUrl(),
         ));
     }
+
+    /**
+     * Match all variables enclosed in "{}" and substitute value using its corresponding attribute.
+     *
+     * @param array $variables  An array with merged defaults variables
+     * @param array $attributes An array of attributes from the matcher
+     *
+     * @return array Attributes with interpreted values
+     */
+    protected function resolveDefaults($variables, $attributes)
+    {
+        $enclosingPattern = "/{(\w+)}/";
+
+        foreach ($variables as $key => $value) {
+            // checking if the variable come from a specified attribute, if so, we shouldn't evaluate because only default values have to be interpreted
+            if (isset($attributes[$key])) {
```

Comment:
@voronkovich Finally, I chose your solution with a little aggregation for checking which variables come from specified attributes